### PR TITLE
Upgrade QMR to Node 20

### PR DESCRIPTION
**Pull request by gmrabian**
### Description

- Upgrade `.nvmrc` to node 20
- Upgrade services to node 20
- Upgrade dependencies that need it for compatibility
- Remove redundant dependencies

Note: there is a lambda that isn't on Node 20. It's created by serverless to link resources as a workaround for...itself. We might be able to narrow down where it comes from with an infrastructure audit, but it just runs once on deploy and then sits there. It'll only be a concern if AWS removes support for Node16, and in that case everyone using serverless will need a solution, so they should upgrade it by then (right...?)

### Related ticket(s)

CMDCT-2659

---
### How to test

Deployed url: https://d1zh5vte8mrjmn.cloudfront.net/
Everything works as expected
Lambdas don't fail

---
### Pre-review checklist

- ~[ ] I have added [thorough](https://shorturl.at/aejkF) tests, if necessary~
- ~[ ] I have updated relevant documentation, if necessary~
- [x] I have performed a self-review of my code
- [ ] I have manually tested this PR in the deployed cloud environment

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-v18.20.2
+v20.11.1
```
Comment (rocio-desantiago) [inline .nvmrc:1]: does this cover `^20.12.7`?

File: package.json [hunk 2]
```diff
@@ -25,7 +25,7 @@
     "@serverless/compose": "^1.3.0",
     "@types/dotenv": "^8.2.0",
     "@types/flat": "^5.0.2",
-    "@types/node": "^18.19.31",
+    "@types/node": "^20.12.7",
```
Comment (rocio-desantiago) [inline package.json:28]: does this version work with .nvmrc having `v20.11.1`?

**Review by rocio-desantiago — APPROVED** [review]
Question about `.nvmrc` file vs `package.json`

File: package.json (hunk 2 above)
Comment (gmrabian) [reply package.json:28]: see above

File: .nvmrc (hunk 1 above)
Comment (rocio-desantiago) [reply .nvmrc:1]: Sweet :shipit:

**Review by ailZhou — APPROVED** [review]

**Pull request merged**